lib/src/main/kotlin/app/cash/kfsm/MachineBuilder.kt

```diff
@@ -84,11 +108,32 @@ class MachineBuilder<ID, V : Value<ID, V, S>, S : State<ID, V, S>> {
     if (this@becomes in transitions) {
       throw IllegalStateException("State $this has multiple `becomes` blocks defined")
     }
-    transitions[this@becomes] = TransitionBuilder<ID, V, S>(this@becomes).apply(block).build()
+    val transitionMap = TransitionBuilder(this@becomes).apply(block).build()
+    val selector: NextStateSelector<ID, V, S> =
+      when (transitionMap.size) {
+        // It's an error to define a becomes block with no subsequent states
+        0 -> throw IllegalStateException("State $this defines a `becomes` block with no transitions")
+        // If there's only one subsequent states, then always select that states if using a selector
+        1 -> NextStateSelector { Result.success(transitionMap.values.first().to) }
```

Comment:
I'm confused by this function returning a `Result.success` but also throwing above instead of returning a `Result.failure`

Reply:
The throw is because the DSL was incorrect. This will make `fsm` return Failure.
The failure is because the user used the state machine incorrectly. This will make `fsm` pass, but return Failure when `machine.advance` is called on a value of the wrong state.

i.e.
```
A becomes {}
```
is Failure.

but

```
val machine = A becomes {
   B via { it }
   C via { it }
} .bind()
machine.transitionTo(valueOfStateA, B) // this is Success
machine.advance(valueOfStateA) // this is Failure because no selector was provided.
```